### PR TITLE
Moved market description to Lua

### PR DESCRIPTION
**Pull request by EPuncker**
### Pull Request Prelude

- [x] I have followed [proper The Forgotten Server code styling][code].
- [x] I have read and understood the [contribution guidelines][cont] before making this PR.
- [x] I am aware that this PR may be closed if the above-mentioned criteria are not fulfilled.

### Changes Proposed
Moved market item description to Lua.
- added "chance to hit" and "range" that have been missing for years
- added ItemType:getWeaponString()
- added itemType:getClassification()

[cont]: https://github.com/otland/forgottenserver/wiki/Contributing
[code]: https://github.com/otland/forgottenserver/wiki/TFS-Coding-Style-Guide

### CODE REVIEW

**Comment by EPuncker** [comment]
yeah it is actually tested and working:
2021-12-18 23_59_59-Tibia - Perflex

**Review by Zbizu — APPROVED** [review]
Let's go. Tier system is waiting.

File: data/scripts/eventcallbacks/player/default_onLookInMarket.lua [hunk 1]
```diff
@@ -0,0 +1,319 @@
...
+			response:addU64(stats.lowestPrice)
+		else
+			response:addByte(0x00)
+		end
+	end
+
+	response:sendToPlayer(self)
+end
+
+ec:register()
```
Comment (DSpeichert) [inline data/scripts/eventcallbacks/player/default_onLookInMarket.lua:319]: This file looks double-indented compared to other Lua files, no?

Comment (Zbizu) [reply data/scripts/eventcallbacks/player/default_onLookInMarket.lua:319]: `do...end` clauses were used to limit the scope of local variables. This has been done in libs already way before this pull request.

Comment (nekiro) [reply data/scripts/eventcallbacks/player/default_onLookInMarket.lua:319]: why though?

Comment (MillhioreBT) [reply data/scripts/eventcallbacks/player/default_onLookInMarket.lua:319]: In this file it is not necessary to use `do ... end` clauses
There is not a single reason for it

**Review by MillhioreBT — CHANGES_REQUESTED** [review]

**Comment by DSpeichert** [comment]
@EPuncker needs rebase

**Comment by EPuncker** [comment]
> @EPuncker needs rebase

fixed.

**Review by DSpeichert — DISMISSED** [review]
2x as much code in Lua as C++, huh?

File: data/scripts/eventcallbacks/player/default_onLookInMarket.lua (hunk 1 above)
Comment (Znote) [reply data/scripts/eventcallbacks/player/default_onLookInMarket.lua:319]: I agree the `do .. end` local scope limitation seems pointless. Is the overhead so big that we shouldn't deconstruct them naturally at the end of function, and force it lots of times throughout the execution? Is this some shenanigans to keep memory overhead as low as possible? I'm not to sure how Lua works, but to me it sounds like your making the function slower by calling the garbage collector way too often. 
And indentation size seems to be set to 8, it should be 4.

**Comment by Zbizu** [comment]
> 2x as much code in Lua as C++, huh?

A few features were missing or had inaccurate descriptions in the C++ version. For example distance weapons show their modifiers now.

File: src/luascript.cpp [hunk 2]
```diff
@@ -42,6 +42,7 @@
 #include "globalevent.h"
 #include "script.h"
 #include "weapons.h"
+#include "iomarket.h"
```
Comment (DSpeichert) [inline src/luascript.cpp:45]: Aren't these alphabetically ordered?

**Review by DSpeichert — CHANGES_REQUESTED** [review]

**Review by DSpeichert — APPROVED** [review]

**Pull request merged**